src/core/types/serialized_enum.rs

```diff
@@ -0,0 +1,228 @@
+use crate::core::types::asset_identifier::AssetIdentifier;
+use crate::core::types::asset_qualifier::AssetQualifier;
+use crate::core::types::scope_spec_identifier::ScopeSpecIdentifier;
+use crate::util::aliases::AssetResult;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+/// There is a bug in cosmwasm 1.0.0's interaction with serde-json-wasm that causes floating point
+/// operations to be added into the compiled wasm, so the previous solution of using things like
+/// AssetIdentifier directly and specifying them with a tag and content param in their serde
+/// annotation is impossible as of 1.0.0.  This solution will allow existing requests to remain
+/// identical, but not generate floating point errors.  It makes the schema less useful, but it's a
+/// hack to fix a bug, so...
+///
+/// It's also worth noting that this solution can only create enum switches that have Strings as
+/// their values.  Anything different will not work for this solution and will require further
+/// adaptation and hackery.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct SerializedEnum {
+    /// Specifies the type of enum to deserialize into.
+    ///
+    /// The word "type" is a reserved keyword in Rust. Need to use serde to interpret this field
+    /// correctly as the desired value of "type"
+    #[serde(rename(serialize = "type", deserialize = "type"))]
+    pub enum_type: String,
```

Comment:
this seems fine, but I think you could also name this `r#type` and it will allow it to be used as `type` https://doc.rust-lang.org/rust-by-example/compatibility/raw_identifiers.html... kind of ugly syntax, but :the_more_you_know: 😄

Reply:
Oh, sweet!  Well, I'd honestly rather do that than use serde annotations to achieve the result, because we all know how safe it is using serde annotations lol.  I'll swap that over and see if it makes the syntax elsewhere zany or not